flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/StandaloneResourceManager.java

```diff
@@ -112,4 +103,21 @@ protected ResourceID workerStarted(ResourceID resourceID) {
 		return resourceID;
 	}
 
+	@Override
+	protected void setFailUnfulfillableRequest(boolean failUnfulfillableRequest) {
+		if (failUnfulfillableRequest) {
+			final long startupPeriodMillis = startupPeriodTime.toMilliseconds();
+
+			if (startupPeriodMillis > 0) {
+				getRpcService().getScheduledExecutor().schedule(
```

Comment:
Let's not use this construct. I think it would be better to simply use `RpcEndpoint#scheduleRunAsync`.